agent/format-2.0.go

```diff
@@ -69,11 +69,12 @@ type format_2_0Serialization struct {
 	QueryTracingEnabled   bool          `yaml:"querytracingenabled,omitempty"`
 	QueryTracingThreshold time.Duration `yaml:"querytracingthreshold,omitempty"`
 
-	OpenTelemetryEnabled     bool   `yaml:"opentelemetryenabled,omitempty"`
-	OpenTelemetryEndpoint    string `yaml:"opentelemetryendpoint,omitempty"`
-	OpenTelemetryInsecure    bool   `yaml:"opentelemetryinsecure,omitempty"`
-	OpenTelemetryStackTraces bool   `yaml:"opentelemetrystacktraces,omitempty"`
-	OpenTelemetrySampleRatio string `yaml:"opentelemetrysampleratio,omitempty"`
+	OpenTelemetryEnabled               bool          `yaml:"opentelemetryenabled,omitempty"`
+	OpenTelemetryEndpoint              string        `yaml:"opentelemetryendpoint,omitempty"`
+	OpenTelemetryInsecure              bool          `yaml:"opentelemetryinsecure,omitempty"`
+	OpenTelemetryStackTraces           bool          `yaml:"opentelemetrystacktraces,omitempty"`
+	OpenTelemetrySampleRatio           string        `yaml:"opentelemetrysampleratio,omitempty"`
+	OpenTelemetryTailSamplingThreshold time.Duration `yaml:"opentelemetrytailsamplingthreshold,omitempty"`
```

Comment:
So when are we starting `format-3.0.go`? :vomiting_face:

Reply:
I regret having multiple values in the config. What I want is a json/yaml file with the options. The only problem there is that it makes the modelling a lot weaker. Although potentially this should be driven by the controller charm.